resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -455,8 +457,13 @@ class ExecutorPodsAllocator(
             .inNamespace(namespace)
             .resource(createdExecutorPod)
             .delete()
-          PVC_COUNTER.decrementAndGet()
+          if (success == 1) {
+            success += 1
+          }
           throw e
+      } finally {
+          val reuse = conf.get(KUBERNETES_DRIVER_OWN_PVC) && conf.get(KUBERNETES_DRIVER_REUSE_PVC)
+          if (!reuse && success == 2) PVC_COUNTER.decrementAndGet()
```

Comment:
`PVC_COUNTER` is only used by SPARK-41410 . `!reuse` is a misleading and useless code, isn't it?

https://github.com/apache/spark/blob/30957a992edb587e8f5046d7c04c848f285688e1/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala#L413

Reply:
The variable `podAllocOnPVC`
```
  private val podAllocOnPVC = conf.get(KUBERNETES_DRIVER_OWN_PVC) &&
    conf.get(KUBERNETES_DRIVER_REUSE_PVC) && conf.get(KUBERNETES_DRIVER_WAIT_TO_REUSE_PVC)
```
includes one more condition for `KUBERNETES_DRIVER_WAIT_TO_REUSE_PVC`.

That's why I think it doesn't hurt to check again.